files/usr/lib/lua/aredn/controller/setup_basic.lua

```diff
@@ -0,0 +1,210 @@
+  Copyright (C) 2021 Darryl Quinn
+  See Contributors file for additional contributors
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation version 3 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  Additional Terms:
+
+  Additional use restrictions exist on the AREDN(TM) trademark and logo.
+    See AREDNLicense.txt for more info.
+
+  Attributions to the AREDN Project must be retained in the source code.
+  If importing this code into a new or existing project attribution
+  to the AREDN project must be added to the source code.
+
+  You must not misrepresent the origin of the material contained within.
+
+  Modified versions must be modified to attribute to the original source
+  and be marked in reasonable ways as differentiate it from the original
+  version.
+
+--]]
+local aredn_info = require("aredn.info")
+local aredn_hardware = require("aredn.hardware")
+require("aredn.utils")
+
+-- -------------------------------------
+-- -- Public API is attached to table
+-- -------------------------------------
+local module = {}
+
+-- Class constructor (NEW method)
+function module:new(req)
+  self.req = req
+  return self
+end
+
+-- Class methods
+function module:process()
+  local res={}
+  
+  if self.req['method']=="GET" then
+    res = module:GET()
+  elseif self.req['method']=="POST" then
+    res = module:POST()
+  else
+    msg="unsupported http method: " .. self.req['method']
+    res['msg']=msg
+    res['success']=false
+  end
+  return res
+end
+
+function module:GET()
+  local res={}
+  local data={}
+  data['basic'] = {}
+  data['basic']['nodename']=aredn_info.getNodeName()
+  data['basic']['description']=aredn_info.getNodeDescription()
+  -- data['password']="WE CANNOT RETRIEVE THE PASSWORD"
+
+  -- MESHRF
+  data['meshrf'] = {}
+  data['meshrf']['ssid_full'] = aredn_info.getSSID()
+  data['meshrf']['ssid_prefix'] = data['meshrf']['ssid_full']:split("-")[1]
+  radio = aredn_info.getMeshRadioDevice()
+  data['meshrf']['enabled'] = aredn_info.isMeshRadioEnabled(radio)
+  data['meshrf']['ip'] = aredn_info.getInterfaceIPAddress("wifi")
+  data['meshrf']['netmask'] = aredn_info.getInterfaceNetmask("wifi")
+  data['meshrf']['distance'] = aredn_info.getMeshRadioDistance(radio)
+  data['meshrf']['bw'] = aredn_info.getChannelBW(radio)
+  data['meshrf']['channel'] = aredn_info.getChannel(radio)
+  data['meshrf']['power'] = aredn_info.getTXPower()
+  data['meshrf']['maxpower'] = aredn_hardware.wifi_maxpower(data['meshrf']['channel'])
+
```

Comment:
How about:

local ssid =  aredn_info.getSSID()
local radio = aredn_info.getMeshRadioDevice()
data.meshrf = {
    ssid_full = ssid,
    ssid_prefix =  ssid:split("-")[1],
    enabled = aredn_info.isMeshRadioEnabled(radio),
    ip = aredn_info.getInterfaceIPAddress("wifi"),
   ... etc ...
}

I think it's a little easier to read (and probably more efficient to construct)